tests/cell_fabric/foo.lef_gold

```diff
@@ -1,7 +1,7 @@
 MACRO foo
-  ORIGIN 0 0 ;
+  ORIGIN 0.0360 0.0520 ;
```

Comment:
This was following your suggestion to set the LEF origin and size correctly so that the placer would properly flip cells with bounding boxes that didn't have lower left corner 0,0. The code was very wrong before if the bbox didn't have origin 0,0. It would say the origin was zero and set the size to the upper right coordinate. Everything should still work if the origin of the bbox is 0,0. The should allow Soner's design to work when the placer is allowed to flip.

Reply:
The placer only accepts primitive cells with bounding boxes that have lower left corner 0, 0.

![image](https://user-images.githubusercontent.com/43042278/77569552-5e09b400-6e98-11ea-80cf-de6fcaa6d472.png)

If you want to define a primitive cell like this, the lower left point should 0, 0, width and height should be w and h. ( the case with lower left poing x,y, width and height w_x and h_y, is not correct.)

Reply:
Why does the placer only accept primitive cells with bounding boxes with a lower left corner of 0,0? In your previous email, you said that it was doing the flipping offset correctly? (Add both llx and urx to the x offset.) If we specify it the above way, you won't know how to offset the cell correctly.

Reply:
What does offset here means?

Flipping correct means, given a placed box with (llx,lly) and (urx,ury), after flipping, the placer was able to give the same (llx,lly) and (urx,ury).

The placer first identify a primitive with lower left point (0,0), width and height. Then the placer give a initial offset to the primitive cell with (llx,lly). Finally, the placer flips the primitive cell according to the (llx,lly), flipping direction, width and height.

Here 'offset' to placer means the initial placed lower left point, which is determined by placer. However, 'offset' in lef file means something else. Does 'offset' int lef  file means some preplacement?

Yaguang

Reply:
How does the placer know what that llx and lly are within the cell? (the point you call x,y in your drawing) Doesn't it need to know that?
Offset means (I think) this x,y point in your drawing. It is not a preplacement. The LEF file is used for a cell template (not cell instance) that can be instantiated in multiple places.

Reply:
The placer can not know the 'offset' inside the cell. The placer flips the cell in cell level, so the 'offset' inside the cell mean nothing to the flip operation (does not need to know 'offset' inside cell).

If you can make sure after cell level flipping, drc is clean, whatever offset inside cell is ok.

Reply:
Was this resolved?

Reply:
@parijatm @stevenmburns @soneryaldiz 

Seems this branch is unnecessary, and can be deleted.